### PR TITLE
allow to set frame file type instead of hardcoded jpg

### PR DESCRIPTION
**Pull request by Shaked**
| Q                  | A
| ------------------ | ---
| Bug fix?           | no
| New feature?       | yes
| BC breaks?         | no
| Deprecations?      | no
| Fixed tickets      | fixes # 
| Related issues/PRs | #493 
| License            | MIT

#### What's in this PR?

Explain the contents of the PR.

When using ExtractMultipleFramesFilter, it would be nice to choose the file type of the frame image e.g jpg, png, etc.. 

#### Why?

Which problem does the PR fix?

It might be better to create frames with png than with jpg: 
https://twitter.com/FFmpeg/status/970382979280154624

#### Example Usage

```php
$frameFileType = 'jpg';
$filter = new ExtractMultipleFramesFilter($frameRate, $destinationFolder);
$filter->setFrameFileType($frameFileType);
```

### CODE REVIEW

File: src/FFMpeg/Filters/Video/ExtractMultipleFramesFilter.php [hunk 1]
```diff
@@ -49,6 +50,10 @@ public function __construct($frameRate = self::FRAMERATE_EVERY_SEC, $destination
         $this->destinationFolder = $destinationFolder;
     }
 
+    public function setFrameFileType($frameFileType) {
```
Comment (jens1o) [inline src/FFMpeg/Filters/Video/ExtractMultipleFramesFilter.php]: please add phpdoc.

File: tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php [hunk 2]
```diff
@@ -34,18 +34,19 @@ public function testApply($frameRate, $destinationFolder, $duration, $modulus, $
...
-            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_30SEC, '/', 100, 2, array('-vf', 'fps=1/30', '/frame-%02d.jpg')),
-            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_60SEC, '/', 100, 2, array('-vf', 'fps=1/60', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/1', '/frame-%03d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_2SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/2', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_5SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/5', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_10SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/10', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_30SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/30', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_60SEC,'png', '/', 100, 2, array('-vf', 'fps=1/60', '/frame-%02d.png')),
         );
```
Comment (jens1o) [inline tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php]: please add tests for `png` and whatever format you can dream of.

Comment (Shaked) [reply tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php]: AFAI can tell only PNG and JPG are supported.

**Comment by jens1o** [comment]
A note for future PRs(in general): Please do not make your changes on your fork's master branch, but instead create a new one. This helps the reviewers, for example when you're filing another pr, while the previous PR was not merged, the changes are included in there.

File: src/FFMpeg/Filters/Video/ExtractMultipleFramesFilter.php [hunk 3]
```diff
@@ -50,6 +50,9 @@ public function __construct($frameRate = self::FRAMERATE_EVERY_SEC, $destination
         $this->destinationFolder = $destinationFolder;
     }
 
+	/**
+	 * @param string $frameFileType
+	 */
     public function setFrameFileType($frameFileType) {
     	$this->frameFileType = $frameFileType;
```
Comment (jens1o) [inline src/FFMpeg/Filters/Video/ExtractMultipleFramesFilter.php]: okay, please check whether `$frameFileType` is one of the extensions defined in a class property with an array of `['jpg', 'jpeg', 'png']`. (`in_array` will help you.)

File: tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php [hunk 4]
```diff
@@ -34,18 +34,25 @@ public function testApply($frameRate, $destinationFolder, $duration, $modulus, $
...
-            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_60SEC, '/', 100, 2, array('-vf', 'fps=1/60', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/1', '/frame-%03d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_2SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/2', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_5SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/5', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_10SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/10', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_30SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/30', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_60SEC,'jpg', '/', 100, 2, array('-vf', 'fps=1/60', '/frame-%02d.jpg')),
+            array(ExtractMultipleFramesFilter::FRAMERATE_EVERY_SEC,'png', '/', 100, 2, array('-vf', 'fps=1/1', '/frame-%03d.png')),
```
Comment (jens1o) [inline tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php]: please insert a line break before this line to improve readability.

Please also put a space before `'jpg'` or `'png'`

File: src/FFMpeg/Filters/Video/ExtractMultipleFramesFilter.php (hunk 3 above)
Comment (Shaked) [reply src/FFMpeg/Filters/Video/ExtractMultipleFramesFilter.php]: Added together with a test

File: tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php [hunk 5]
```diff
@@ -41,18 +41,34 @@ public function testApply($frameRate, $frameFileType,$destinationFolder, $durati
...
         );
     }
+
+    /**
+     * @expectedException \FFMpeg\Exception\InvalidArgumentException
+     */
+    public function testInvalidFrameFileType() {
+    	$filter = new ExtractMultipleFramesFilter('1/1', '/');
+        $filter->setFrameFileType('webm');
+    }
```
Comment (jens1o) [inline tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php]: at least one line is indented incorrectly.

**Comment by jens1o** [comment]
will need to test this, hopefully tomorrow I've got time to play around with it. Please be patient.

File: tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php (hunk 5 above)
Comment (Shaked) [reply tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php]: Sorry for that, fixed. Might be worth using some PHP formatter in the future

Comment (jens1o) [reply tests/Unit/Filters/Video/ExtractMultipleFramesFilterTest.php]: Yeah, that's the plan for the `v1.x`-branch. No worries :)

**Comment by jens1o** [comment]
Works good!

**Pull request merged**